pkg/messages/interfaces.go

```diff
@@ -0,0 +1,41 @@
+package messages
+
+import (
+	"encoding"
+	"github.com/numaproj/numaflow/pkg/isb"
+	"github.com/numaproj/numaflow/pkg/watermark/processor"
+	"time"
+)
+
+// NumaMessage is the representation of a message flowing through the numaflow
+type NumaMessage interface {
+	ReadMessage
+	encoding.BinaryMarshaler
+	encoding.BinaryUnmarshaler
+	// SetID sets the ID for a message
+	SetID(id string)
+	// ID is the unique identifier for a message
+	ID() string
+	// SetEventTime represents the timestamp of the message depending on the time characteristics of the pipeline.
+	// please note that EventTime does not mean only the event time.
+	// This is intended to be set once and should be immutable. Therefore, an error could be thrown if
+	// an attempt is made to overwrite the time on a message.
+	SetEventTime(t time.Time) error
```

Comment:
I don't think we need to worry about errors. we can assume that the error checks are handled by the framework itself (in other words, only good data is used to call setters).